dataset-workflows/no2-so2-cdk/cdk/cdk_stack.py

```diff
@@ -0,0 +1,185 @@
+            runtime=aws_lambda.Runtime.FROM_IMAGE,
+            memory_size=1024,
+            timeout=core.Duration.seconds(30),
+        )
+
+        s3_discovery_lambda.add_to_role_policy(
+            aws_iam.PolicyStatement(
+                actions=["s3:GetObject"],
+                resources=[f"arn:aws:s3:::{bucket}/*"],
+            )
+        )
+        s3_discovery_lambda.add_to_role_policy(
+            aws_iam.PolicyStatement(
+                actions=["s3:ListBucket"],
+                resources=[f"arn:aws:s3:::{bucket}"],
+            )
+        )
+
+       
+
+
+
+        generate_stac_item_lambda = aws_lambda.Function(
+            self,
+            f"{id}-{collection}-generate-stac-item-fn",
+            code=aws_lambda.Code.from_asset_image(
+                directory="../../lambdas/stac-gen",
+                file="Dockerfile",
+                entrypoint=["/usr/local/bin/python", "-m", "awslambdaric"],
+                cmd=["handler.handler"],
+            ),
+            handler=aws_lambda.Handler.FROM_IMAGE,
+            runtime=aws_lambda.Runtime.FROM_IMAGE,
+            memory_size=4096,
+            timeout=core.Duration.seconds(60),
+            environment=dict(
+                EARTHDATA_USERNAME=os.environ["EARTHDATA_USERNAME"],
+                EARTHDATA_PASSWORD=os.environ["EARTHDATA_PASSWORD"],
+            ),
+        )
+        generate_stac_item_lambda.add_to_role_policy(full_bucket_access)
+
+        db_write_role = aws_iam.Role(
+            self,
+            "PGStacLoaderRole",
+            assumed_by=aws_iam.ServicePrincipal("lambda.amazonaws.com"),
+            managed_policies=[
+                aws_iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "service-role/AWSLambdaBasicExecutionRole"
+                )
+            ],
+        )
+
+
+        db_write_lambda = aws_lambda.Function(
+            self,
+            f"{id}-{collection}-write-db-fn",
+            role=db_write_role,
+            code=aws_lambda.Code.from_asset_image(
+                directory="../../lambdas/db-write",
+                file="Dockerfile",
+                entrypoint=["/usr/local/bin/python", "-m", "awslambdaric"],
+                cmd=["handler.handler"],
+            ),
+            handler=aws_lambda.Handler.FROM_IMAGE,
+            runtime=aws_lambda.Runtime.FROM_IMAGE,
+            memory_size=4096,
+            timeout=core.Duration.seconds(60),
+            environment=dict(
+                STAC_DB_HOST=os.environ["STAC_DB_HOST"],
+                STAC_DB_USER=os.environ["STAC_DB_USER"],
+                STAC_DB_PASSWORD=os.environ["STAC_DB_PASSWORD"],
+            ),
+            vpc=database_vpc,
+            vpc_subnets=ec2.SubnetSelection(subnet_type=ec2.SubnetType.PRIVATE),
+            security_groups=[lambda_function_security_group],
+        )
+
+        db_write_lambda.add_to_role_policy(ec2_network_access)
+
+        s3_start_state = stepfunctions.Pass(self, "S3 Discovery StartState")
```

Comment:
As far as I can tell, this is being executed manually right? At one point I thought we discussed triggering this class of workflow whenever there was new data in the `s3://climatedashboard-data/delivery/<COLLECTION_NAME>` location but it looks like we're kicking it off manually here and discovering data to publish in `s3://climatedashboard-data/<COLLECTION_NAME>`

Reply:
at the moment yes, this is executed manually. i agree it would be good to add a trigger, i can look into it.

Reply:
cool, not a priority though, I would add it as a todo in a comment or open an issue